core/templates/pages/contributor-dashboard-page/opportunities-list/opportunities-list.component.ts

```diff
@@ -16,110 +16,120 @@
+    new EventEmitter()
+  );
+  @Input() opportunityHeadingTruncationLength: number;
 
-      ctrl.directiveSubscriptions.add(
-        TranslationLanguageService.onActiveLanguageChanged.subscribe(
-          () => ctrl.$onInit()));
+  loadingOpportunityData: boolean;
+  lastPageNumber: number;
+  opportunities: ExplorationOpportunity[] = [];
+  visibleOpportunities = [];
+  directiveSubscriptions = new Subscription();
+  activePageNumber: number;
+  OPPORTUNITIES_PAGE_SIZE = constants.OPPORTUNITIES_PAGE_SIZE;
 
-      ctrl.directiveSubscriptions.add(
-        ContributionOpportunitiesService
-          .reloadOpportunitiesEventEmitter.subscribe(() => ctrl.$onInit()));
+  constructor(
+    private readonly contributionOpportunitiesService:
+      ContributionOpportunitiesService,
+    private readonly translationLanguageService: TranslationLanguageService) {
+    this.directiveSubscriptions.add(
+      this.translationLanguageService.onActiveLanguageChanged.subscribe(
+        () => this.ngOnInit()));
 
-      ctrl.directiveSubscriptions.add(
-        ContributionOpportunitiesService
-          .removeOpportunitiesEventEmitter.subscribe((opportunityIds) => {
-            opportunities = opportunities.filter(function(opportunity) {
-              return opportunityIds.indexOf(opportunity.id) < 0;
-            });
-            ctrl.visibleOpportunities = opportunities.slice(
-              0, OPPORTUNITIES_PAGE_SIZE);
-            ctrl.lastPageNumber = Math.ceil(
-              opportunities.length / OPPORTUNITIES_PAGE_SIZE);
-          }));
+    this.directiveSubscriptions.add(
+      this.contributionOpportunitiesService
+        .reloadOpportunitiesEventEmitter.subscribe(() => this.ngOnInit()));
 
-      ctrl.$onDestroy = function() {
-        ctrl.directiveSubscriptions.unsubscribe();
-      };
+    this.directiveSubscriptions.add(
+      this.contributionOpportunitiesService
+        .removeOpportunitiesEventEmitter.subscribe((opportunityIds) => {
+          this.opportunities = this.opportunities.filter((opportunity) => {
+            return opportunityIds.indexOf(opportunity.id) < 0;
+          });
+          this.visibleOpportunities = this.opportunities.slice(
+            0, this.OPPORTUNITIES_PAGE_SIZE);
+          this.lastPageNumber = Math.ceil(
+            this.opportunities.length / this.OPPORTUNITIES_PAGE_SIZE);
+        }));
+  }
 
-      ctrl.$onInit = function() {
-        opportunities = [];
-        ctrl.visibleOpportunities = [];
-        ctrl.activePageNumber = 1;
-        ctrl.lastPageNumber = 1000;
-        ctrl.loadingOpportunityData = true;
+  ngOnDestroy(): void {
+    this.directiveSubscriptions.unsubscribe();
+  }
 
-        ctrl.loadOpportunities().then(function({opportunitiesDicts, more}) {
-          opportunities = opportunitiesDicts;
-          // "more" returned from GAE storage is not always reliable if true.
+  ngOnInit(): void {
+    this.opportunities = [];
+    this.visibleOpportunities = [];
```

Comment:
Can these lines be removed if they are already being initialized above?

Reply:
Done